Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -3,67 +3,97 @@
 
 /* eslint-disable react/display-name */
 import React, { useState, useEffect, useMemo, useContext, useCallback } from 'react';
-import { LgEditor, LGOption } from '@bfc/code-editor';
+import { LgEditor } from '@bfc/code-editor';
 import get from 'lodash/get';
 import debounce from 'lodash/debounce';
 import isEmpty from 'lodash/isEmpty';
-import { LgFile } from '@bfc/indexers';
 import { editor } from '@bfcomposer/monaco-editor/esm/vs/editor/editor.api';
-import { lgIndexer, Diagnostic, combineMessage, isValid } from '@bfc/indexers';
+import { lgIndexer, combineMessage, isValid } from '@bfc/indexers';
+import { RouteComponentProps } from '@reach/router';
+import querystring from 'query-string';
 
 import { StoreContext } from '../../store';
 import * as lgUtil from '../../utils/lgUtil';
 
 const { check } = lgIndexer;
 
-interface CodeEditorProps {
-  file: LgFile;
-  template: lgUtil.Template | null;
-  line: number;
-}
-
 const lspServerPath = '/lg-language-server';
 
-export default function CodeEditor(props: CodeEditorProps) {
-  const { actions } = useContext(StoreContext);
-  const { file, template, line } = props;
+interface CodeEditorProps extends RouteComponentProps<{}> {
+  fileId?: string;
+}
+
+const CodeEditor: React.FC<CodeEditorProps> = props => {
+  const { actions, state } = useContext(StoreContext);
+  const { lgFiles } = state;
+  const { fileId } = props;
+  const file = lgFiles?.find(({ id }) => id === 'common');
   const [diagnostics, setDiagnostics] = useState(get(file, 'diagnostics', []));
   const [content, setContent] = useState('');
   const [errorMsg, setErrorMsg] = useState('');
   const [lgEditor, setLgEditor] = useState<editor.IStandaloneCodeEditor | null>(null);
 
-  const fileId = file && file.id;
+  const search = props?.location?.search ?? '';
+  const searchTemplateName = querystring.parse(search).t;
+  const templateId = Array.isArray(searchTemplateName) ? searchTemplateName[0] : searchTemplateName;
+  const template = templateId && file && file.templates.find(({ name }) => name === templateId);
+
+  const hash = props?.location?.hash;
+  const lineMatched = hash && /L(\d+)/g.exec(hash);
```

Comment:
Can we make the fragment that can be parsed by the query-string package? ex: `#L=50`